homeassistant/components/ecobee/number.py

```diff
@@ -0,0 +1,93 @@
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the ecobee thermostat number entity."""
+    data = hass.data[DOMAIN]
+    entities = []
+    _LOGGER.debug("Adding min time ventilators numbers (if present)")
+    for index in range(len(data.ecobee.thermostats)):
+        thermostat = data.ecobee.get_thermostat(index)
+        if thermostat["settings"]["ventilatorType"] != "none":
+            _LOGGER.debug("Adding %s's ventilator min times number", thermostat["name"])
+            entities.append(
+                EcobeeVentilatorMinTime(
+                    data,
+                    index,
+                    "home",
+                    "ventilatorMinOnTimeHome",
+                    data.ecobee.set_ventilator_min_on_time_home,
+                )
+            )
+
+            entities.append(
+                EcobeeVentilatorMinTime(
+                    data,
+                    index,
+                    "away",
+                    "ventilatorMinOnTimeAway",
+                    data.ecobee.set_ventilator_min_on_time_away,
+                )
+            )
+
+    async_add_entities(entities, True)
+
+
+class EcobeeVentilatorMinTime(NumberEntity):
+    """A number class, representing min time  for an ecobee thermostat with ventilator attached."""
+
+    _attr_native_min_value = 0
+    _attr_native_max_value = 60
+    _attr_native_step = 5
+    _attr_native_unit_of_measurement = UnitOfTime.MINUTES
+    _attr_has_entity_name = True
+
+    def __init__(self, data, thermostat_index, mode, ecobee_setting_key, set_func):
+        """Initialize ecobee ventilator platform."""
+        self.data = data
+        self.thermostat_index = thermostat_index
+        self.thermostat = self.data.ecobee.get_thermostat(self.thermostat_index)
+        self.ecobee_setting_key = ecobee_setting_key
+        self.set_func = set_func
+        self._attr_name = f"Ventilator min time {mode}"
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, self.thermostat["identifier"])},
+            manufacturer=MANUFACTURER,
+            model=ECOBEE_MODEL_TO_NAME.get(self.thermostat["modelNumber"]),
+            name=self.thermostat["name"],
+        )
+        self._attr_unique_id = f'{self.thermostat["identifier"]}_ventilator_{mode}'
+        self._attr_native_value = self.thermostat["settings"][ecobee_setting_key]
+
+    async def async_update(self):
+        """Get the latest state from the thermostat."""
+        await self.data.update()
+        self.thermostat = self.data.ecobee.get_thermostat(self.thermostat_index)
+        self._attr_native_value = self.thermostat["settings"][self.ecobee_setting_key]
+
+    @property
+    def available(self):
+        """Return if device is available."""
+        return self.thermostat["runtime"]["connected"]
```

Comment:
It looks like this could be moved into a base entity class as well

```
binary_sensor.py:        return thermostat["runtime"]["connected"]
climate.py:        return self.thermostat["runtime"]["connected"]
climate.py:        """Return true if humidifier connected to thermostat and set to manual/on mode."""
humidifier.py:        return self.thermostat["runtime"]["connected"]
number.py:        return self.thermostat["runtime"]["connected"]
sensor.py:        return thermostat["runtime"]["connected"]
```